auth/src/main/java/com/firebase/ui/auth/AuthUI.java

```diff
@@ -403,17 +407,111 @@ public static AuthUI getInstance(FirebaseApp app) {
+            mProviderId = providerId;
+        }
+
+        public String getProviderId() {
+            return mProviderId;
+        }
+
+        public List<String> getScopes() {
+            return mScopes;
+        }
+
+        protected IdpConfig(Parcel in) {
+            mProviderId = in.readString();
+            mScopes = in.createStringArrayList();
+        }
+
+        public static final Creator<IdpConfig> CREATOR = new Creator<IdpConfig>() {
+            @Override
+            public IdpConfig createFromParcel(Parcel in) {
+                return new IdpConfig(in);
+            }
+
+            @Override
+            public IdpConfig[] newArray(int size) {
+                return new IdpConfig[size];
+            }
+        };
+
+        @Override
+        public int describeContents() {
+            return 0;
+        }
+
+        @Override
+        public void writeToParcel(Parcel parcel, int i) {
+            parcel.writeString(mProviderId);
+            parcel.writeStringList(mScopes);
+        }
+
+        public static class Builder {
+            private String mProviderId;
+            private List<String> mScopes;
+
+
+            /**
+             * Builds the configuration parameters for an identity provider.
+             * @param providerId An ID of one of the supported identity providers. e.g.
+             * {@link AuthUI#GOOGLE_PROVIDER}. See {@link AuthUI#SUPPORTED_PROVIDERS} for the
+             * complete list of supported Identity providers
+             */
+            public Builder(@NonNull String providerId) {
+                if (!SUPPORTED_PROVIDERS.contains(providerId)) {
+                    throw new IllegalArgumentException("Unkown provider: " + providerId);
+                }
+                mProviderId = providerId;
+            }
+
+            /**
+             * Specifies the additional permissions that the application will request for this
+             * identity provider.
+             *
+             * For Facebook permissions see:
+             * https://developers.facebook.com/docs/facebook-login/android
+             * https://developers.facebook.com/docs/facebook-login/permissions
+             *
+             * For Google permissions see:
+             * https://developers.google.com/identity/protocols/googlescopes
+             */
+            public Builder setPermissions(List<String> permissions) {
+                mScopes = permissions;
+                return this;
+            }
+
+            public IdpConfig build() {
+                return new IdpConfig(mProviderId, mScopes);
+            }
+        }
+    }
+
     /**
      * Builder for the intent to start the user authentication flow.
      */
     public final class SignInIntentBuilder {
         private int mLogo = NO_LOGO;
         private int mTheme = getDefaultTheme();
-        private List<String> mProviders = Collections.singletonList(EMAIL_PROVIDER);
+        private List<IdpConfig> mProviders = new ArrayList<>();
```

Comment:
We do need set like properties, but we also need to retain the order.  It looks like LinkedHashSet will do what we want.